src/modules/logicModule/LM_PC_FundingPot_v1.sol

```diff
@@ -0,0 +1,133 @@
+    // State
+
+    /// @dev The role for the funding pot admin.
+    bytes32 public constant FUNDING_POT_ADMIN_ROLE = "FUNDING_POT_ADMIN";
+
+    /// @notice    Storage gap for future upgrades.
+    uint[50] private __gap;
+
+    // =========================================================================
+    // Constructor & Init
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        __Module_init(orchestrator_, metadata_);
+    }
+
+    // =========================================================================
+    // Public - Mutating
+
+    /// @inheritdoc ILM_PC_FundingPot_v1
+    /// @notice Grants the funding pot admin role to the given address.
+    /// @dev This function is only callable by the orchestrator admin.
+    /// @param admin_ The address to grant the funding pot admin role to.
+    function grantFundingPotAdminRole(address admin_)
+        external
+        onlyOrchestratorAdmin
+    {
+        if (_checkForFundingPotAdminRole(admin_)) {
+            revert Module__LM_PC_FundingPot_FundingPotAdminAlreadySet();
+        }
+        __Module_orchestrator.authorizer().grantRole(
+            getFundingPotAdminRoleId(), admin_
+        );
+    }
+
+    /// @inheritdoc ILM_PC_FundingPot_v1
+    /// @notice Revokes the funding pot admin role from the given address.
+    /// @dev This function is only callable by the orchestrator admin.
+    /// @param admin_ The address to revoke the funding pot admin role from.
+    function revokeFundingPotAdminRole(address admin_)
+        external
+        onlyOrchestratorAdmin
+    {
+        if (!_checkForFundingPotAdminRole(admin_)) {
+            revert Module__LM_PC_FundingPot_AddressIsNotFundingPotAdmin();
+        }
+        __Module_orchestrator.authorizer().revokeRole(
+            getFundingPotAdminRoleId(), admin_
+        );
+    }
+
+    // =========================================================================
+    // Public - Getters
+
+    /// @notice Generates a role id for the funding pot admin role.
+    function getFundingPotAdminRoleId() public view returns (bytes32) {
+        return __Module_orchestrator.authorizer().generateRoleId(
+            address(this), FUNDING_POT_ADMIN_ROLE
+        );
+    }
+    //--------------------------------------------------------------------------
+    // Internal
+
+    /// @dev    Checks if the given address has the funding pot admin role.
+    /// @param  admin_ The address to check for the funding pot admin role.
+    /// @return bool True if the address has the funding pot admin role, false otherwise.
+    function _checkForFundingPotAdminRole(address admin_)
+        internal
+        view
+        returns (bool)
+    {
+        return __Module_orchestrator.authorizer().checkForRole(
+            getFundingPotAdminRoleId(), admin_
+        );
+    }
```

Comment:
All of this is handled by the authorizer, so you don't have to bother about it :)

To be more precise:
- the orchestrator admin can call `grantModuleRole` on the funding pot module because that one is inheriting from Module_v1 (like [here](https://github.com/InverterNetwork/contracts/blob/template/logic-module-payment-client/src/templates/tests/unit/LM_PC_Template_v1.t.sol#L155))
- `grantModuleRole` does the assigning of the role
- you only need to define the role as a constant (which you did)
- then, any function that needs to be gated to be accessible only by that role just needs the modifier `onlyModuleRole` (like [here](https://github.com/InverterNetwork/contracts/blob/template/logic-module-payment-client/src/templates/modules/LM_PC_Template_v1.sol#L185C9-L185C23))